File: src/prime.cpp
#include <iostream>
#include <math.h>
#include <stdlib.h>

using namespace std;

bool isPrime(int i){
	if(i%2==0) return false;
	for(int j=3;j<((int) sqrt(i))+1;j+=2){
		if(i%j==0) return false;
	}
	return true;
}

int main(int argc, char * argv[]){

	if(argc==1){
		cout << "Requires one argument, remember target" << endl;
		return 0;
	}

	int sum=0, target=atoi(argv[1]);

	for (int i=2;i<=target;i++){
		if(isPrime(i)){
			
			sum++;

			//cout << i << endl;
		}
	
	}
	
	cout << sum << endl;
}

File: src/prime.rb
def isPrime(n)
  if n<2 
    return false
  end
  if n == 2
    return true
  end
  if n%2==0
    return false
  end
  i=3
  while i<Math.sqrt(n)+1
    if n % i == 0
      return false
    end
    i+=2
  end

  return true
end

if ARGV.length != 1 
  puts "requires one argument, remember target"
  exit(0)
end

target = ARGV[0].to_i

counter = 0
for i in 1..target
  if isPrime(i)
    counter+=1
  end
end
puts counter
